src/include/interval_set.h

```diff
@@ -722,6 +722,8 @@ class interval_set {
    */
   void move_into(Map& other) {
     other = std::move(m);
+    m.clear();
```

Comment:
i am not sure if we need this change. i think the better approach is to drop this method. and use 

```c++
m = std::move(im);
```

for two reasons:

- better readability, as the convention of the semantic of "move" is well-known. the caller does not need to refer to the implementation for understanding what `move_into()` implies.
- less maintenance overhead. we don't need to document this function or fix it anymore.

also, i don't think it's necessary to fix this method. as we cannot assume that a moved away instance is cleared. and it's the caller's responsibility to do the cleanup, if it wants to make sure it's cleared after moving away from the object.

Reply:
> i am not sure if we need this change. i think the better approach is to drop this method. and use
> 
> ```c++
> m = std::move(im);
> ```

I don't see how that's possible. The types are different? I could see this:

````
Map detach() {
  return std::move(m);
}
````

> for two reasons:
> 
>     * better readability, as the convention of the semantic of "move" is well-known. the caller does not need to refer to the implementation for understanding what `move_into()` implies.

I suppose that can be replaced by the caller manually calling `im.clear();`.

Reply:
> > i am not sure if we need this change. i think the better approach is to drop this method. and use
> > ```c++
> > m = std::move(im);
> > ```
> 
> I don't see how that's possible. The types are different? I could see this:

sorry, i misread the code.

> 
> ```
> Map detach() {
>   return std::move(m);
> }
> ```
> 
> > for two reasons:
> > ```
> > * better readability, as the convention of the semantic of "move" is well-known. the caller does not need to refer to the implementation for understanding what `move_into()` implies.
> > ```
> 
> I suppose that can be replaced by the caller manually calling `im.clear();`.

yeah. i am for this change.

Reply:
Done